common/djangoapps/third_party_auth/tests/test_models.py

```diff
@@ -17,39 +25,61 @@ def setUp(self):
         super().setUp()
         self.saml_provider_config = SAMLProviderConfigFactory()
 
-    def test_unique_entity_id_enforcement_for_non_current_configs(self):
-        """
-        Test that the unique entity ID enforcement does not apply to noncurrent configs
-        """
-        with self.assertLogs() as ctx:
-            assert len(SAMLProviderConfig.objects.all()) == 1
-            old_entity_id = self.saml_provider_config.entity_id
-            self.saml_provider_config.entity_id = f'{self.saml_provider_config.entity_id}-ayylmao'
-            self.saml_provider_config.save()
-
-            # check that we now have two records, one non-current
-            assert len(SAMLProviderConfig.objects.all()) == 2
-            assert len(SAMLProviderConfig.objects.current_set()) == 1
-
-            # Make sure we can use that old entity id
-            SAMLProviderConfigFactory(entity_id=old_entity_id)
-
-            # 7/21/22 : Disabling the exception on duplicate entity ID's because of existing data.
-            # with pytest.raises(IntegrityError):
-            bad_config = SAMLProviderConfig(entity_id=self.saml_provider_config.entity_id)
-            bad_config.save()
-        assert ctx.records[0].msg == f'Entity ID: {self.saml_provider_config.entity_id} already in use'
-
-    @override_settings(FEATURES={'ENABLE_UNICODE_USERNAME': False})
-    def test_clean_username_unicode_disabled(self):
-        """
-        Test the username cleaner function with unicode disabled
-        """
-        assert clean_username('ItJüstWòrks™') == 'ItJ_stW_rks'
+    @ddt.data(
+        ('ItJüstWòrks™', False, 'ItJ_stW_rks'),
+        ('ItJüstWòrks™', True, 'ItJüstWòrks'),
+        ('simple_username', False, 'simple_username'),
+        ('simple_username', True, 'simple_username'),
+        ('test@example.com', False, 'test_example_com'),
+        ('test@example.com', True, 'test@example.com'),
+    )
+    @ddt.unpack
+    def test_clean_username(self, input_username, unicode_enabled, expected_output):
+        """Test the username cleaner function with different unicode settings."""
+        with override_settings(FEATURES={'ENABLE_UNICODE_USERNAME': unicode_enabled}):
+            self.assertEqual(clean_username(input_username), expected_output)
+
 
-    @override_settings(FEATURES={'ENABLE_UNICODE_USERNAME': True})
-    def test_clean_username_unicode_enabled(self):
+@ddt.ddt
+class TestSAMLConfigurationSignals(TestCase):
```

Comment:
1. We lost all testing/assertions for custom_attributes. See https://github.com/openedx/edx-platform/pull/36954#discussion_r2229575556
2. This class should live in `signals/tests/test_handlers.py` if we follow the conventions for the rest of the tests.
3. Since we made no `models.py` changes, I suggest you make no `test_models.py` changes in this PR either. If you feel strongly about adjusting these tests for some reason, you can do that in a separate PR. Thanks.